code/modules/antagonists/traitor/datum_traitor.dm

```diff
@@ -23,43 +7,93 @@
 	antag_moodlet = /datum/mood_event/focused
 	antag_hud_type = ANTAG_HUD_TRAITOR
 	antag_hud_name = "traitor"
-	hijack_speed = 0.5 //10 seconds per hijack stage by default
-	var/employer = "The Syndicate"
+	///10 seconds per hijack stage by default
+	hijack_speed = 0.5
+	ui_name = "TraitorInfo"
+	///give this traitor objectives? doesn't include ending objective
 	var/give_objectives = TRUE
-	var/should_give_codewords = TRUE
-	var/should_equip = TRUE
+	///give this traitor codewords? nanotrasen traitors do not.
+	var/give_codewords = TRUE
+	///give this traitor an uplink?
+	var/give_uplink = TRUE
+	///if TRUE, this traitor will always get hijacking as their final objective
+	var/is_hijacker = FALSE
+
+	///the name of the antag flavor this traitor has.
+	var/employer
+
+	///assoc list of strings set up after employer is given
+	var/list/traitor_flavor
 
+	///datum of the contractor hub, if they decide to become a contractor in the round.
+	///in the future, this should definitely be moved into a component that attaches to these datums
 	var/datum/contractor_hub/contractor_hub
 
+	///reference to the uplink this traitor was given, if they were.
+	var/datum/component/uplink/uplink
+
+	///how many more objectives past the starting bunch need to get finished before escape objective is unlocked
+	var/additional_objectives_before_escape = 4
+	///if new objectives won't be added because a previously finished objective got unfinished, lazylist reference to uncompleted objectives
+	var/list/new_objectives_blocked
+	///when new objectives are blocked, the objectives to add pile up here
+	var/objectives_to_add = 0
+	///the final objective the traitor has to accomplish, be it escaping, hijacking, or just martyrdom.
+	var/datum/objective/ending_objective
+
 /datum/antagonist/traitor/on_gain()
 	owner.special_role = job_rank
 	if(give_objectives)
 		forge_traitor_objectives()
+	//will be given later
+	//forge_ending_objective()
+	for(var/datum/objective/smart/objective in objectives)
+		RegisterSignal(objective, COMSIG_SMART_OBJECTIVE_ACHIEVED, .proc/objective_done)
+		RegisterSignal(objective, COMSIG_SMART_OBJECTIVE_UNACHIEVED, .proc/objective_undone)
+
+
+	var/faction = prob(75) ? FACTION_SYNDICATE : FACTION_NANOTRASEN
```

Comment:
🤯

Reply:
so you're saying define it, right?

Reply:
actually my mind was blown about nanotrasen traitors

Reply:
is a nanotrasen traitor a traitor to the syndicate or what